sympy/printing/lambdarepr.py

```diff
@@ -336,6 +336,31 @@ def _print_Function(self, e):
                                 func_name)
         return "%s(%s)" % (nstr, self._print_seq(e.args))
 
+    def _print_Piecewise(self, expr):
+        "Piecewise function printer"
+        exprs = [self._print(arg.expr) for arg in expr.args]
+        conds = [self._print(arg.cond) for arg in expr.args]
+        # If [default_value, True] is a (expr, cond) sequence in a Piecewise object
+        #     it will behave the same as passing the 'default' kwarg to select()
+        #     *as long as* it is the last element in expr.args.
+        # If this is not the case, it may be triggered prematurely.
+        ans = []
+        parenthesis_count = 0
+        is_last_cond_True = False
+        for cond, expr in zip(conds, exprs):
+            if cond == 'True':
+                ans.append(expr)
+                is_last_cond_True = True
+                break
+            else:
+                ans.append('where(%s,%s,' % (cond, expr))
+                parenthesis_count += 1
+        if not is_last_cond_True:
+            # simplest way to put a nan but raises
+            # 'RuntimeWarning: invalid value encountered in log'
+            ans.append('log(-1)')
```

Comment:
I am not an expert of numexpr but found no other way to produce nan that does not depend on the environment (you could of course define np_nan = np.nan and use it in the numexpr expression but it would cause problems if a variable is called np_nan).

Indeed, numexpr.evaluate('log(complex(-1, 0))') returns array(3.141592653589793j), but I don't think you can configure numexpr to work with log(-1). Anyway we choose its configuration in the printer. The warning is just ugly. Maybe someone can catch this specific warning and yield another? Anyway, not using True as final condition is not recommended.